lusidtools/cocoon/transaction_type_upload.py

```diff
@@ -58,3 +58,91 @@ def create_transaction_type_configuration(api_factory, alias, movements):
     )
 
     return response
+
+
+def replace_current_transaction_alias(
+    api_factory, updated_alias, movement, properties=None
+):
+
+    """
+    This function overrides the current transaction types alias with a new configuration
+
+    Parameters
+    ----------
+    api_factory: lusid.utilities.ClientApiFactory
+        The LUSID api factory to use.
+    updated_alias: lusid.models.TransactionConfigurationTypeAlias
+        The alias which we want to update.
+    movements: list[lusid.models.TransactionConfigurationMovementDataRequest]
+        The movements to use for transaction type
+    properties:
+
+    Returns
+    -------
+    response : (lusid.models.TransactionSetConfigurationData)
+        The response from setting the transaction type
+    """
+
+    # Call LUSID to get a list of the current transaction types
+
+    current_transaction_types = api_factory.build(
+        lusid.api.SystemConfigurationApi
+    ).list_configuration_transaction_types()
+
+    transaction_configs_list = current_transaction_types.transaction_configs
+
+    # Build the new transaction type you want to upload
+
+    new_transaction_request = models.TransactionConfigurationDataRequest(
+        aliases=[updated_alias], movements=movement, properties=properties,
+    )
+
+    # Delete the current configuration assigned to your replacement alias
+
+    for txn_index, trans_type in enumerate(transaction_configs_list):
+
+        find_new_alias = False
+
+        for alias_index, alias in enumerate(trans_type.aliases):
+
+            if (
+                alias.type == updated_alias.type
+                and alias.transaction_group == updated_alias.transaction_group
+            ):
+
+                del trans_type.aliases[alias_index]
+
+                find_new_alias = True
+
+                break
+            else:
+                continue
+        else:
+            continue
+        break
```

Comment:
This is nice, so is this saying that if the condition...

``` 
if (
                alias.type == updated_alias.type
                and alias.transaction_group == updated_alias.transaction_group
            ):
```

Is not met at any time in the for loop, then it will continue to the next transaction type, however if it has been met then it will break as there is no need to look at any more transaction types?

Reply:
Exactly.
Once found, the rest of the loop is redundant. 
There should only be one unique pair of **alias_type and transaction_group** in each LUSID.